docs/src/setup-script-generation-plugin/index.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+var PrebuildPlugin = require("prebuild-webpack-plugin");
+const fs = require("fs");
+const glob = require("glob");
+const yaml = require("js-yaml");
+const Mustache = require("mustache");
+
+function generateSetupScripts() {
+  return glob("../app/boards/**/*.zmk.yml", (error, files) => {
+    const aggregated = files.flatMap((f) =>
+      yaml.safeLoadAll(fs.readFileSync(f, "utf8"))
+    );
+
+    const data = aggregated.reduce(
+      (agg, item) => {
+        switch (item.type) {
+          case "shield":
+            item.compatible = true;
+            item.split = item.siblings?.length > 1;
+            agg.keyboards.push(item);
+            break;
+          case "board":
+            if (!item.features?.includes("keys")) {
+              agg.boards.push(item);
+            }
+            break;
+        }
+        return agg;
+      },
+      { keyboards: [], boards: [] }
+    );
+
+    const templateBuffer = fs.readFileSync(
+      "src/templates/setup.sh.mustache",
+      "utf8"
+    );
+    const script = Mustache.render(templateBuffer, data);
+    fs.writeFileSync("static/setup.sh", script);
+    const powershellTemplateBuffer = fs.readFileSync(
+      "src/templates/setup.ps1.mustache",
+      "utf8"
+    );
+    const psScript = Mustache.render(powershellTemplateBuffer, data);
+    return fs.writeFileSync("static/setup.ps1", psScript);
```

Comment:
Minor: not much of an issue with only two scripts, but you could define an array of the scripts to generate and loop through them, since the render code is the same for them all aside from template file and output file.